src/overlays/actors/ovl_En_Vm/z_en_vm.c

```diff
@@ -27,35 +33,480 @@ const ActorInit En_Vm_InitVars = {
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Vm/func_80B2D460.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Vm/EnVm_Init.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_METAL_SHIELD, 0x00, 0x0D, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 25, 70, 0, { 0, 0, 0 } },
+};
+
+static ColliderQuadInit sQuadInit1 = {
+    { COLTYPE_METAL_SHIELD, 0x11, 0x00, 0x00, 0x00, COLSHAPE_QUAD },
+    { 0x00, { 0xFFCFFFFF, 0x00, 0x10 }, { 0x00000000, 0x00, 0x00 }, 0x81, 0x00, 0x00 },
+    { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
+};
+
+static ColliderQuadInit sQuadInit2 = {
+    { COLTYPE_METAL_SHIELD, 0x00, 0x09, 0x00, 0x00, COLSHAPE_QUAD },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+    { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
+};
+
+static Vec3f D_80B2EAEC = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EAF8 = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB04 = { 500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB10 = { -500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB1C = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB28 = { 0.0f, 0.0f, 1600.0f };
+
+static Vec3f D_80B2EB34 = { 1000.0f, 700.0f, 2000.0f };
+
+static Vec3f D_80B2EB40 = { 1000.0f, -700.0f, 2000.0f };
+
+static Vec3f D_80B2EB4C = { -1000.0f, 700.0f, 1500.0f };
+
+static Vec3f D_80B2EB58 = { -1000.0f, -700.0f, 1500.0f };
+
+static Vec3f D_80B2EB64 = { 500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB70 = { -500.0f, 0.0f, 0.0f };
+
+static Vec3f D_80B2EB7C = { 0.4f, 0.4f, 0.4f };
+
+static UNK_PTR D_80B2EB88[] = { 0x0402CFE0, 0x0402D7E0, 0x0402DFE0, 0x0402E7E0, 0x0402EFE0,
+                                0x0402F7E0, 0x0402FFE0, 0x040307E0, 0x04030FE0, 0x040317E0 };
+
+extern SkeletonHeader D_06003F60;
+extern AnimationHeader D_06000068;
+extern Gfx D_04031FE0[];
+extern Gfx D_06002728[];
+
+void EnVm_SetupAction(EnVm* this, EnVmActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnVm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnVm* this = THIS;
+
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06003F60, &D_06000068, this->limbDrawTable,
+                   this->transitionDrawTable, 11);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 0.0f);
+    Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+    Collider_SetCylinder(globalCtx, &this->colliderCylinder, thisx, &sCylinderInit);
+    Collider_InitQuad(globalCtx, &this->colliderQuad1);
+    Collider_SetQuad(globalCtx, &this->colliderQuad1, thisx, &sQuadInit1);
+    Collider_InitQuad(globalCtx, &this->colliderQuad2);
+    Collider_SetQuad(globalCtx, &this->colliderQuad2, thisx, &sQuadInit2);
+    this->beamSightRange = (thisx->params >> 8) * 40.0f;
+    thisx->params &= 0xFF;
+    thisx->naviEnemyId = 0x39;
```

Comment:
was mixing thisx and this required to match?

Reply:
yeah it was